Linking fails for missing LDFLAGS -lm

This has originally been reported here: https://github.com/nethappen/blocksync-fast/issues/9

But I still can reproduce it on Fedora 43/44.

Here's how to reproduce. Copy the following to a `Dockerfile`:
```
FROM fedora:44

RUN dnf install -y git gcc libgcrypt-devel automake
RUN git clone https://github.com/nethappen/blocksync-fast.git /root/blocksync-fast
WORKDIR /root/blocksync-fast
RUN autoreconf --install
RUN ./configure

CMD ["make"]
```

and then:
```
docker build --tag fedora-blocksync . && docker run fedora-blocksync
make  all-recursive
make[1]: Entering directory '/root/blocksync-fast'
Making all in src
make[2]: Entering directory '/root/blocksync-fast/src'
gcc -DHAVE_CONFIG_H -I. -I..       -g -O2 -MT blocksync-fast.o -MD -MP -MF .deps/blocksync-fast.Tpo -c -o blocksync-fast.o blocksync-fast.c
mv -f .deps/blocksync-fast.Tpo .deps/blocksync-fast.Po
gcc -DHAVE_CONFIG_H -I. -I..       -g -O2 -MT utils.o -MD -MP -MF .deps/utils.Tpo -c -o utils.o utils.c
mv -f .deps/utils.Tpo .deps/utils.Po
gcc -DHAVE_CONFIG_H -I. -I..       -g -O2 -MT globals.o -MD -MP -MF .deps/globals.Tpo -c -o globals.o globals.c
mv -f .deps/globals.Tpo .deps/globals.Po
gcc -DHAVE_CONFIG_H -I. -I..       -g -O2 -MT common.o -MD -MP -MF .deps/common.Tpo -c -o common.o common.c
mv -f .deps/common.Tpo .deps/common.Po
gcc -DHAVE_CONFIG_H -I. -I..       -g -O2 -MT init.o -MD -MP -MF .deps/init.Tpo -c -o init.o init.c
mv -f .deps/init.Tpo .deps/init.Po
gcc -DHAVE_CONFIG_H -I. -I..       -g -O2 -MT benchmark.o -MD -MP -MF .deps/benchmark.Tpo -c -o benchmark.o benchmark.c
mv -f .deps/benchmark.Tpo .deps/benchmark.Po
gcc -DHAVE_CONFIG_H -I. -I..       -g -O2 -MT digest_info.o -MD -MP -MF .deps/digest_info.Tpo -c -o digest_info.o digest_info.c
mv -f .deps/digest_info.Tpo .deps/digest_info.Po
gcc    -g -O2   -o blocksync-fast blocksync-fast.o utils.o globals.o common.o init.o benchmark.o digest_info.o -lgcrypt  
/usr/bin/ld.bfd: blocksync-fast.o: in function `print_progress':
/root/blocksync-fast/src/blocksync-fast.c:191:(.text+0x1fb): undefined reference to `floorf'
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:377: blocksync-fast] Error 1
make[2]: Leaving directory '/root/blocksync-fast/src'
make[1]: *** [Makefile:393: all-recursive] Error 1
make: *** [Makefile:334: all] Error 2
make[1]: Leaving directory '/root/blocksync-fast'
```

--- a/configure.ac
+++ b/configure.ac
@@ -27,7 +27,7 @@ AC_CONFIG_HEADERS([config.h])
 AC_PROG_CC
 
 # Checks for libraries.
-#AC_CHECK_LIB([m], [main])
+AC_SEARCH_LIBS([floor], [m])
 
 # Checks for header files.
 AC_CHECK_HEADERS([fcntl.h stdint.h sys/param.h unistd.h])
@@ -41,7 +41,7 @@ AC_TYPE_SIZE_T
 AC_FUNC_MALLOC
 AC_FUNC_MMAP
 AC_FUNC_REALLOC
-AC_CHECK_FUNCS([ftruncate getpagesize munmap strcasecmp strerror strtoul])
+AC_CHECK_FUNCS([floor ftruncate getpagesize munmap strcasecmp strerror strtoul])
 
 # Checking for libgcrypt
 LIBGCRYPT_MIN_VERSION=1.9.0

